app/models/application_record.rb

```diff
@@ -1,5 +1,25 @@
 class ApplicationRecord < ActiveRecord::Base
   self.abstract_class = true
 
-  after_initialize
+  after_initialize :send_notification
+  after_commit :clear_cache
+
+  private
+
+  def send_notification
+    return unless persisted?
+
+    ActiveSupport::Notifications.instrument(
+      'initialized.active_record',
+      record: self
+    )
+  end
+
+  def clear_cache
+    Rails.cache.redis.smembers(self.cache_key).each do |cached_fragment|
```

Comment:
This probably doesn't work, the record's cache key is already updated. We need to queue a cache busting job with the cache key before it's updated, or else make the cache key not change across updates.